BaseUnit.cpp

```diff
@@ -0,0 +1,24 @@
+#include "BaseUnit.h"
+
+BaseUnit::BaseUnit(const std::string& nm, int hp, int dmg) : Name{ nm }, HP{hp}, DMG{dmg}
+{
+}
+
+//decreasing the HP of the suffering unit, return with false if it dies
+bool BaseUnit::gotHit(int dam)
```

Comment:
Jelenlegi állapotban még ez is jó megoldás lenne, de kicsit továbbgondolva az osztályok feladatkörét, OO szempontból jobb lenne, ha paraméterben a támadó unitot kapná meg, és az alapján számolná ki a sebzést. Így pl. később jól meg lehet különböztetni, hogy a sebzést egy másik unit okozta, vagy egy környezeti hatás, méreg, spell, stb.
A másik, hogy 1 függvény 1 feladatot lásson el. Azt, hogy halott, ne (vagy ne csak) ez adja vissza, hanem erre legyen egy külön `isDead` metódus.

Reply:
Kész van a változtatás.
Elég lenne itt válaszolgatnom a javaslatokra, vagy kérjek review-t?

Reply:
Elég ha review-t kérsz, abból tudom, hogy kész, és átnézhetem.